src/Behat/Behat/Transformation/Transformer/FromStringObjectTransformer.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/*
+ * This file is part of the Behat.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Behat\Behat\Transformation\Transformer;
+
+use Behat\Behat\Definition\Call\DefinitionCall;
+use Behat\Behat\Transformation\Exception\FactoryMethodNotFound;
+use Closure;
+use ReflectionClass;
+use ReflectionParameter;
+
+/**
+ * Transforms typehinted parameters using their `fromString` factory method (if exists).
+ *
+ * @author Konstantin Kudryashov <ever.zet@gmail.com>
+ */
+final class FromStringObjectTransformer implements ArgumentTransformer
+{
+    const FACTORY_METHOD = 'fromString';
```

Comment:
Could this be passed into the constructor instead? It could come from configuration but have a default?